base/number.jl

```diff
@@ -3,6 +3,15 @@
 ## generic operations on numbers ##
 
 isinteger(x::Integer) = true
+iszero(x::Number) = x == 0 # faster than x == zero(x) for complex, bigfloat, etc.
```

Comment:
Technically, comparing a unitful number to `0` is still a well-defined operation...